### ISSUE TITLE
[BUG] custom grains not working with salt-call

### ISSUE THREAD

**Issue by ysagon** [saltstack/salt#65873]
**Description**
Since the update to 3006.5 custom grains aren't working with salt-call. They still do work with salt 'my_host'.

**Setup**
my custom grains are in `/srv/salt/state/_grains/`

- [x] on-prem machine
- [x] onedir packaging

**Steps to Reproduce the behavior**
No grains returned, launched from `admin1.baobab`
```
(baobab)-[root@admin1 state] (master *)$ salt-call grains.get mellanox_card
```
Grains returned, launched from `admin1.baobab`
```
(baobab)-[root@admin1 state] (master *)$ salt 'admin1.baobab'  grains.get mellanox_card
admin1.baobab:
    ConnectX-5
```

**Expected behavior**
same grains when using salt and salt-call

I did previously:
```
salt "admin1.baobab" saltutil.sync_grains
```
and 
```
salt-call saltutil.sync_grains
```

**Versions Report**
<details><summary>salt --versions-report</summary>
(Provided by running salt --versions-report. Please also mention any differences in master/minion versions.)

```yaml
(baobab)-[root@admin1 state] (master *)$ salt --versions-report
Salt Version:
          Salt: 3006.5

Python Version:
        Python: 3.10.13 (main, Nov 15 2023, 04:34:27) [GCC 11.2.0]

Dependency Versions:
          cffi: 1.14.6
      cherrypy: unknown
      dateutil: 2.8.1
     docker-py: 7.0.0
         gitdb: Not Installed
     gitpython: Not Installed
        Jinja2: 3.1.2
       libgit2: Not Installed
  looseversion: 1.0.2
      M2Crypto: Not Installed
          Mako: Not Installed
       msgpack: 1.0.2
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     packaging: 22.0
     pycparser: 2.21
      pycrypto: Not Installed
  pycryptodome: 3.9.8
        pygit2: Not Installed
  python-gnupg: 0.4.8
        PyYAML: 6.0.1
         PyZMQ: 23.2.0
        relenv: 0.14.2
         smmap: Not Installed
       timelib: 0.2.4
       Tornado: 4.5.3
           ZMQ: 4.3.4

System Versions:
          dist: rocky 8.8 Green Obsidian
        locale: utf-8
       machine: x86_64
       release: 4.18.0-477.27.1.el8_8.x86_64
        system: Linux
       version: Rocky Linux 8.8 Green Obsidian
```
</details>

**Comment by ysagon** [comment]
I tried with our previous version 3006.3 it is working as expected.

**Comment by ysagon** [comment]
I tried to keep salt-master to version 3006.5 and downgrade the minion to 3006.4 and it is working as well.

**Comment by max-arnold** [comment]
The issue is already tracked here https://github.com/saltstack/salt/issues/65027

**Comment by ysagon** [comment]
Or this one? https://github.com/saltstack/salt/issues/65692

**Comment by Foorack** [comment]
Yup, #65692 and was merged in #65738. Will likely go into next 3006.x, so only solution for now is to downgrade from to 3006.4.

**Comment by ysagon** [comment]
Thanks, I'm closing this issue as it is already tracked.
